File: 백준/Silver/10799. 쇠막대기/쇠막대기.kt

import java.util.Stack

fun main(){
    val br = System.`in`.bufferedReader()
    val stack = Stack<Char>()
    val pipes = br.readLine()
    var ret = 0
    for(i in 0 until pipes.length){
        if(pipes[i]=='('){
            stack.push('(')
        }else{
            stack.pop()
            if(pipes[i-1]==')'){
               ret += 1
            }else{
                ret += stack.size
            }
        }
    }
    println(ret)
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys

input = sys.stdin.readline
#
ps = list(input().strip())
stack = []
count = 0

for i, ch in enumerate(ps):
    if ch == '(':
        stack.append('(')
    else:
        stack.pop()
        if ps[i - 1] == '(':  # 레이저
            count += len(stack)
        else:  # 막대기 끝
            count += 1

print(count)
